frontend/javascripts/oxalis/view/right-border-tabs/segments_tab/segments_view.tsx

```diff
@@ -1842,6 +1953,49 @@ class SegmentsView extends React.Component<Props, State> {
     );
   }
 
+  getExpandSubgroupsItem(groupId: number) {
+    const children: Key[] = this.getSubGroupsAsTreeNodes(groupId);
+    const expandedGroupsSet = new Set(this.state.expandedGroupKeys);
+    const areAllChildrenExpanded = children.every((childNode) => expandedGroupsSet.has(childNode));
+    const isGroupItselfExpanded = expandedGroupsSet.has(getKeyForGroupId(groupId));
+    if (areAllChildrenExpanded && isGroupItselfExpanded) {
+      return null;
+    }
+    const allExpandedGroups = children.concat(this.state.expandedGroupKeys ?? []);
+    if (!isGroupItselfExpanded) allExpandedGroups.push(getKeyForGroupId(groupId));
+    return {
+      key: "expandAll",
+      disabled: !this.props.allowUpdate,
+      onClick: () => {
+        this.expandGroups(allExpandedGroups);
+        this.closeSegmentOrGroupDropdown();
+      },
+      icon: <ExpandAltOutlined />,
+      label: "Expand all subgroups",
+    };
+  }
+
+  getCollapseSubgroupsItem(groupId: number) {
+    const children = this.getSubGroupsAsTreeNodes(groupId);
+    const expandedKeySet = new Set(this.state.expandedGroupKeys);
+    const areAllChildrenCollapsed =
+      children.filter((childNode) => expandedKeySet.has(childNode)).length === 0;
```

Comment:
```suggestion
    const areAllChildrenCollapsed =
      children.every((childNode) => !expandedKeySet.has(childNode));
```

`every` and `some` are faster on average because they can abort the iteration early as soon as the predicate cannot change anymore.